data-management/neo4j-service.js

```diff
@@ -174,6 +176,45 @@ async function listArms(parameters) {
 }
 
 //Mutations
+async function requestArmAccess(armIDs, userInfo) {
+    const arms = Array.isArray(armIDs) ? armIDs : [armIDs];
+    const promises = arms.map(async (arm) => {
+        const aArm = ArmAccess.createRequestAccess();
+        const cypher =
+            `
+            MATCH (user:User) WHERE user.email='${userInfo.email}' and user.IDP ='${userInfo.idp}'
+            OPTIONAL MATCH (arm:Arm) WHERE arm.armID=$armID
+            CREATE (user)<-[:of_user]-(access:Access {
+                accessStatus: '${aArm.getAccessStatus()}',
+                requestDate: '${getTimeNow()}',
+                requestID: '${aArm.getRequestID()}'
+            })-[:of_arm]->(arm)
+            RETURN access
+            `
+        return await executeQuery({armID: arm}, cypher, 'access');
+    });
+    return await Promise.all(promises);
+}
+
+// Searching for valid arms excluding approved or requested arm
+async function searchValidRequestArm(parameters, user) {
+    const cypher =
+        `
+        MATCH (user:User)-[*..1]-(req:Access)-[*..1]-(userArm:Arm)
+        WHERE user.email='${user.getEmail()}' and user.IDP ='${user.getIDP()}' and req.accessStatus in ['${REQUESTED}', '${APPROVED}']

```

Comment:
if a user requests 'requested' or 'approved' as an access-arm status again, it will throw an error.